src/tests/ftest/util/server_utils.py

```diff
@@ -108,18 +90,17 @@ def create_server_yaml(basepath, log_filename):
 
     # Update values from avocado_testcase.yaml in DAOS yaml variables.
     if new_value_set:
-        for key in new_value_set['server_config']:
-            if key in default_value_set['servers'][0]:
+        if 'server' in new_value_set['server_config']:
```

Comment:
Might I suggest a slightly cleaner approach?  Also the 'key' is a string so it is best to use `==` or `!=` rather than `in` or `not in`.

```
    for key, value in new_value_set['server_config'].items():
        if key == 'server' and isinstance(value, dict):
            # Add or replace any server yaml default 'server' value with the test yaml value
            for server_key, server_value in value.items():
                default_value_set['servers'][0][server_key] = server_value
        elif key in default_value_set['servers'][0]:
            # Replace any other server yaml default 'server' value with the test yaml value
            default_value_set['servers'][0][key] = value
        elif key in default_value_set:
            # Replace any other server yaml default value with the test yaml value
            default_vlaue_set[key] = value
```